voice/udp/manager.go

```diff
@@ -160,7 +165,13 @@ func (m *Manager) ReadPacket() (p *Packet, err error) {
 		return nil, ErrManagerClosed
 	}
 
-	return conn.ReadPacket()
+	p, err = conn.ReadPacket()
+	if err != nil && conn.IsClosed() {
+		wsutil.WSDebug("UDP connection was closed, re-attempting read...")
+		return m.ReadPacket()
```

Comment:
I think it's a better idea to make this a loop. Using recursion for this seems overkill.

Reply:
tfw no tail call optimizatoin, sure.